src/hooks/Trades.test.ts

```diff
@@ -0,0 +1,34 @@
+import { BASES_TO_CHECK_TRADES_AGAINST, CUSTOM_BASES } from '../constants'
+import { generateAllRoutePairs } from './Trades'
+import { ChainId } from '@uniswap/sdk'
+
+describe('Trade', () => {
+  const chainId = ChainId.MAINNET
+  const baseTokens = BASES_TO_CHECK_TRADES_AGAINST[chainId]
+  const restrictedTokens = CUSTOM_BASES[chainId]
+
+  it('should find only restricted pairs for restricted-route tokens', () => {
+    for (const restrictedToken in restrictedTokens) {
```

Comment:
Tests should not contain this much logic.

It is hard to understand exactly what is expected of the function by reading this test.

Better to try just a few inputs under different conditions and check their exact outputs.

eg test function with two random tokens, 1 token from common bases and 1 not, both from common bases, 1 from restricted 1 not, 2 from restricted, etc.

Each of these would be separate test cases